src/combinatorics/simple_graph/connectivity/subgraph.lean

```diff
@@ -0,0 +1,213 @@
+/-
+Copyright (c) 2023 Kyle Miller, Rémi Bottinelli. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kyle Miller, Rémi Bottinelli
+-/
+
+import combinatorics.simple_graph.basic
+import combinatorics.simple_graph.connectivity
+import combinatorics.simple_graph.subgraph
+/-!
+# Connectivity of subgraphs
+-/
+
+universes u v
+
+namespace simple_graph
+
+variables {V : Type u} {V' : Type v} {G : simple_graph V} {G' : simple_graph V'}
+
+/-- A subgraph is connected if it is connected as a simple graph. -/
+@[reducible] def subgraph.connected (H : G.subgraph) : Prop := H.coe.connected
+
+lemma subgraph.connected_iff (H : G.subgraph) :
+  H.connected ↔ H.coe.preconnected ∧ H.verts.nonempty :=
+begin
+  change H.coe.connected ↔ _,
+  rw [connected_iff, set.nonempty_coe_sort],
+end
```

Comment:
Since you changed `subgraph.connected` to a reducible def, you can do this now:
```suggestion
by rw [subgraph.connected, connected_iff, set.nonempty_coe_sort]
```